src/cache_client.rs

```diff
@@ -62,30 +62,15 @@ impl CacheClient {
     /// use momento::requests::cache::create_cache::CreateCache;
     /// # let (cache_client, cache_name) = create_doctest_cache_client();
     ///
-    /// let create_cache_response = cache_client.create_cache(cache_name).await?;
-    ///
-    /// assert_eq!(create_cache_response, CreateCache::AlreadyExists {});
+    /// match cache_client.create_cache(&cache_name).await? {
+    ///     CreateCache::Created => println!("Cache {} created", &cache_name),
+    ///     CreateCache::AlreadyExists => println!("Cache {} already exists", &cache_name),
+    /// }
     /// # Ok(())
     /// # })
     /// # }
     /// ```
-    /// You can also use the [send_request](CacheClient::send_request) method to create a cache using a [CreateCacheRequest]:
-    /// ```no_run
-    /// # fn main() -> anyhow::Result<()> {
-    /// # use momento_test_util::create_doctest_cache_client;
-    /// # tokio_test::block_on(async {
-    /// use momento::requests::cache::create_cache::CreateCache;
-    /// use momento::requests::cache::create_cache::CreateCacheRequest;
-    /// # let (cache_client, cache_name) = create_doctest_cache_client();
-    ///
-    /// let create_cache_request = CreateCacheRequest::new(cache_name);
-    ///
-    /// let create_cache_response = cache_client.send_request(create_cache_request).await?;
-    ///
-    /// assert_eq!(create_cache_response, CreateCache::AlreadyExists {});
-    /// # Ok(())
-    /// # })
-    /// # }
+    /// You can also use the [send_request](CacheClient::send_request) method to create a cache using a [CreateCacheRequest].
```

Comment:
if we are going to move examples out of these docs, we need to make sure we make it really obvious that the send_request path provides optional arguments that are not available in the shorthand API. In addition to adding verbiage to that effect I almost think that we should enumerate the optional arguments from the request object here. It's important that users know why they would want to opt in to that alternate syntax.

@malandis any thoughts?

Reply:
Agreed. If we presently have optional arguments and building the request is the way to do it, that should be in the examples. If at present there are no optional arguments, then it's less important, but if we change the request at some point to add optional arguments, we should demonstrate it in the docs.